### PR TITLE
Perplexity Provider. 

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR:

- solves issue #21

### CODE REVIEW

File: prem_utils/models.json
Comment (Anindyadeep) [inline prem_utils/models.json:1]: Since, now I am adding a specific class for each of the providers (inheriting from OpenAIconnector, so do we need to specify "perplexity/codellama-34b-instruct" here? 

Also, another reason I feel, the list_models function is required, so that we can avoid model redundancy by giving additonal prefix specific to the providers in models.json :)

Comment (filopedraz) [reply prem_utils/models.json:1]: Here you need to specify a unique slug otherwise there will be conflicts in the models.json, then in the corresponding provider, you need to manipulate the slug in order to reflect what is accepted by the APIs of the specific provider.

**Review by filopedraz — APPROVED** [review]

**Comment by Anindyadeep** [comment]
I will merge this, after PR #47 get's merged and if this looks okay now :)

File: prem_utils/models.json
Comment (Anindyadeep) [reply prem_utils/models.json:1]: Done

**Pull request merged**
